lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2206,5 +2212,68 @@ def test_data_bool_not_eq(self):
         self.assertFalse(cube1 == cube2)
 
 
+class Test__eq__meta(tests.IrisTest):
+    def test_ancillary_fail(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        avr = AncillaryVariable([2, 3], long_name="foo")
+        cube2.add_ancillary_variable(avr, 0)
+        self.assertFalse(cube1 == cube2)
+
+    def test_ancillary_reorder(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        avr1 = AncillaryVariable([2, 3], long_name="foo")
+        avr2 = AncillaryVariable([4, 5], long_name="bar")
+        # Add the same ancillary variables to cube1 and cube2 in
+        # opposite orders.
+        cube1.add_ancillary_variable(avr1, 0)
+        cube1.add_ancillary_variable(avr2, 0)
+        cube2.add_ancillary_variable(avr2, 0)
+        cube2.add_ancillary_variable(avr1, 0)
+        self.assertTrue(cube1 == cube2)
+
+    def test_cell_measure_fail(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        cms = CellMeasure([2, 3], measure="area", long_name="foo")
+        cube2.add_cell_measure(cms, 0)
+        self.assertFalse(cube1 == cube2)
+
+    def test_cell_measure_reorder(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        cms1 = CellMeasure([2, 3], measure="area", long_name="foo")
+        cms2 = CellMeasure([4, 5], measure="area", long_name="bar")
+        # Add the same cell measure to cube1 and cube2 in
+        # opposite orders.
+        cube1.add_cell_measure(cms1, 0)
+        cube1.add_cell_measure(cms2, 0)
+        cube2.add_cell_measure(cms2, 0)
+        cube2.add_cell_measure(cms1, 0)
+        self.assertTrue(cube1 == cube2)
+
+    def test_cell_method_fail(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        cmth = CellMethod("mean", "time", "6hr")
+        cube2.add_cell_method(cmth)
+        self.assertFalse(cube1 == cube2)
+
+    # Unlike cell measures, cell methods are order sensitive.
+    def test_cell_method_reorder_fail(self):
+        cube1 = Cube([0, 1])
+        cube2 = Cube([0, 1])
+        cmth1 = CellMethod("mean", "time", "6hr")
+        cmth2 = CellMethod("mean", "time", "12hr")
+        # Add the same cell method to cube1 and cube2 in
+        # opposite orders.
+        cube1.add_cell_method(cmth1)
+        cube1.add_cell_method(cmth2)
+        cube2.add_cell_method(cmth2)
+        cube2.add_cell_method(cmth1)
+        self.assertFalse(cube1 == cube2)
```

Comment:
Should we have a **successful** comparison with cell-methods here, or is that covered somewhere else ?

Reply:
This is capturing the behaviour mentioned in the comments on #3530. Adding cell_methods in different orders _should_ cause equality to fail.

Reply:
Misunderstanding I think !
What I meant to say was :  do we need another testcase showing that equality testing succeeds, when the cubes have cell methods, but they *do* match.

Reply:
ah, yeah, that sounds appropriate.